objects/src/accounts/storage/mod.rs

```diff
@@ -49,6 +50,30 @@ impl AccountStorage {
         Ok(Self { slots })
     }
 
+    // TODO Document.
+    pub fn from_components(
+        components: &[AccountComponent],
+    ) -> Result<AccountStorage, AccountError> {
+        let storage_slots = components
+            .iter()
+            .flat_map(|component| component.storage_slots())
+            .cloned()
+            .collect();
+
+        Self::new(storage_slots)
+    }
+
+    pub fn from_components_with_faucet_metadata(
+        components: &[AccountComponent],
+        faucet_metadata: Word,
+    ) -> Result<AccountStorage, AccountError> {
+        let mut storage_slots = vec![StorageSlot::Value(faucet_metadata)];
+        storage_slots
+            .extend(components.iter().flat_map(|component| component.storage_slots()).cloned());
+
+        Self::new(storage_slots)
+    }
```

Comment:
Potentially, I'd move this logic into the account constructor.

Reply:
I agree that would be nice, but given the dependence of the AccountId on the code and storage, I don't know how we could efficiently rewrite places like these:

https://github.com/0xPolygonMiden/miden-base/blob/4cf01859dcf1fd798f2b9566de6e855634691c5e/miden-lib/src/accounts/faucets/mod.rs#L105-L116

We can of course call `Account::from_components` on the same components again, but that will redo work we've just done (including `MastForest::merge` which is expensive).

Reply:
I introduced `Account::initialize_from_components` to get away from initializing code and storage separately and also to have a central place to validate supported types of the components, but `AccountCode::from_components` is still useful for testing. We could try to make it private though and only call it from `initialize_from_components` and expose a public version of it behind `testing` for tests.

Reply:
I think this is fine for now and we can revisit this in a follow-up PR.